test/rt-test.py

```diff
@@ -79,42 +26,33 @@ def cleanup():
     os.killpg(os.getpgid(proc.pid), signal.SIGTERM)
 
 if __name__ == "__main__":
-    arch = sys.argv[1]
-    zig_path = sys.argv[2]
-    spec = importlib.util.spec_from_file_location("arch", "test/kernel/arch/" + arch + "/rt-test.py")
-    arch_module = importlib.util.module_from_spec(spec)
-    spec.loader.exec_module(arch_module)
+    zig_path = sys.argv[1]
+
+    proc = subprocess.Popen(zig_path + " build run -Drt-test=true", stdout=subprocess.PIPE, shell=True, preexec_fn=os.setsid)
+    atexit.register(cleanup)
+    case_idx = 0
+    read_thread = threading.Thread(target=read_messages, args=(proc,))
+    read_thread.daemon = True
+    read_thread.start()
 
-    # The list of log statements to look for before arch init is called +
-    # All log statements to look for, including the arch-specific ones +
-    # The list of log statements to look for after arch init is called
-    cases = get_pre_archinit_cases() + arch_module.get_test_cases(TestCase) + get_post_archinit_cases()
+    finished_test = False
+
+    while True:
+        try:
+            line = msg_queue.get(block=True, timeout=5)
+        except queue.Empty:
+            if (finished_test):
+                print("\nTesting Successful")
+                break
+            failure("Timed out")
+
+        if "FAILURE" in line:
+            failure(line)
+        elif "[ERROR] Kernel panic:" in line:
+            # As the panic test will be last, then the timeout can elapse
+            finished_test = True
```

Comment:
What if we get a panic that isn't the one we expected? Is that possible? I also noticed that you have to wait 5 seconds (the timeout length) after the tests have been run until the program exits. Can you think of a way to make the program exit as soon as all tests have been run and were successful? If not that's fine, we can get back to it if it gets annoying.

Reply:
I will have another think :)